java/src/Ice/src/main/java/com/zeroc/Ice/ConnectionI.java

```diff
@@ -1443,8 +1443,6 @@ private void setState(int state) {
               }
             }
             // else don't resume reading since we're at or over the _maxDispatches limit.
-
-            _threadPool.register(this, SocketOperation.Read);
```

Comment:
Bug in the MaxDispatches implementation (now fixed). Detected by the updated hold test.